### PR TITLE
Add testing for cl_d3d10.h, cl_d3d11.h, and cl_dx9_media_sharing.h

### PR DESCRIPTION
**Pull request by bashbaug**
Adds missing testing for cl_d3d10.h, cl_d3d11.h, and cl_dx9_media_sharing.h.

### CODE REVIEW

**Review by alycm — DISMISSED** [review]
LGTM.

**Comment by bashbaug** [comment]
This PR has been open for a while too.  Are any additional changes needed?  If not, can it be approved and merged?  Thanks!

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -19,6 +22,9 @@ endfunction(add_header_test)
 
 # Tests
 add_header_test(cl_h test_cl.h.c)
+add_header_test(cl_d3d10_h test_cl_d3d10.h.c)
+add_header_test(cl_d3d11_h test_cl_d3d11.h.c)
+add_header_test(cl_dx9_media_sharing_h test_cl_dx9_media_sharing.h.c)
```
Comment (kpet) [inline tests/CMakeLists.txt:27]: Maybe worth adding `cl_dx9_media_sharing_intel.h` while it's there.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -2,9 +2,12 @@
 function(add_header_test NAME SOURCE)
   foreach(VERSION 100 110 120 200 210 220 300)
     set(TEST_EXE ${NAME}_${VERSION})
-    # ICD on Windows uses system headers, which aren't strictly ANSI conformant
-    # and trigger warnings 
-    if(NOT (NAME STREQUAL cl_icd_h AND MSVC AND CMAKE_C_FLAGS MATCHES "/Za"))
+    # Some headers include system files which require anonymous structs
+    # so we cannot test them with /Za:
```
Comment (kpet) [inline tests/CMakeLists.txt:6]: That's a more useful comment, thanks!

**Review by kpet — COMMENTED** [review]
Looks good but we should probably add a test for `cl_dx9_media_sharing_intel.h` while we're at it.

**Review by kpet — APPROVED** [review]
LGTM

**Comment by bashbaug** [comment]
I've rebased this after the layer's change.  I think this can go in anytime.  Thanks!

**Review by jrprice — APPROVED** [review]

**Pull request merged**